### PR TITLE
test: Add edge case tests for etcd codecs

### PR DESCRIPTION
**Pull request by ahrav**
Adds comprehensive tests for edge cases in the etcd codec implementations.
This includes testing with:
- Maximum field values for owner_value.
- Empty key ranges and metadata for shard_record.
- Zero operation log entries for shard_record.
- Maximum values for run and shard IDs.
- Rejection of trailing bytes during decoding.
- Truncated blobs during decoding.
- Maximum field size for vector reads.
- Zero-filled payloads for vector reads.

## Summary by CodeRabbit

* **Improvements**
  * Replaced panic/unsafe constructors with fallible, validated APIs; unified size limits and schema constants.
  * Redacted debug output to avoid exposing raw bytes; deterministic IDs, fixed-size encodings, and zero-allocation critical paths enforced.
  * Errors improved with user-friendly display and forward-extensible variants.

* **Tests**
  * Expanded unit and property tests covering encoding/decoding, size bounds, round-trips, edge cases, and invariant enforcement.

### CODE REVIEW

File: crates/gossip-coordination-etcd/src/codec_tests.rs [hunk 1]
```diff
@@ -853,6 +895,36 @@ fn decoder_read_vec_valid() {
...
+    push_u32(&mut buf, MAX_FIELD_SIZE as u32);
+    buf.extend_from_slice(&payload);
+
+    let mut d = Decoder::new(&buf);
+    assert_eq!(d.read_vec().unwrap(), payload);
+    assert!(
+        d.finish().is_ok(),
+        "decoder should consume the full payload"
+    );
+}
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-coordination-etcd/src/codec_tests.rs]: Partially addressed. The suggestion to use a 1-byte payload does not work — `read_vec` reads the length prefix then consumes exactly `len` bytes from the buffer, so a prefix of `MAX_FIELD_SIZE` with fewer bytes yields `Truncated`, not a boundary acceptance test. The allocation is required to exercise the `len > MAX_FIELD_SIZE` guard at equality.

However, the full `assert_eq!(d.read_vec().unwrap(), payload)` was doing an O(1 MiB) byte-by-byte comparison unnecessary for a boundary condition test. Replaced with a length-only check (`assert_eq!(result.unwrap().len(), MAX_FIELD_SIZE)`) to eliminate the comparison cost while keeping the allocation that the test fundamentally requires.
